src/Plugin/TfaSetup/TfaTotpSetup.php

```diff
@@ -0,0 +1,186 @@
+<?php
+
+namespace Drupal\ga_login\Plugin\TfaSetup;
+
+use Base32\Base32;
+use Drupal\Core\Link;
+use Drupal\tfa\Plugin\TfaSetupInterface;
+use Drupal\tfa\Plugin\TfaValidation\TfaTotp;
+use Drupal\Core\Url;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\user\UserDataInterface;
+use Drupal\user\Entity\User;
+
+/**
+ * @TfaSetup(
+ *   id = "tfa_totp_setup",
+ *   label = @Translation("TFA Totp Setup"),
+ *   description = @Translation("TFA Totp Setup Plugin"),
+ *   help_links = {
+ *    "Google Authenticator (Android/iPhone/BlackBerry)" = "https://support.google.com/accounts/answer/1066447?hl=en",
+ *    "Authy (Android/iPhone)" = "https://www.authy.com/thefuture#install-now",
+ *    "FreeOTP (Android)" = "https://play.google.com/store/apps/details?id=org.fedorahosted.freeotp",
+ *    "GAuth Authenticator (desktop)" = "https://github.com/gbraad/html5-google-authenticator"
+ *   }
+ * )
+ */
+class TfaTotpSetup extends TfaTotp implements TfaSetupInterface {
```

Comment:
I have planned it for this week to explore the options. I was reading up on design patterns and had some ideas related to integrate decorators in other class but havent given this one much thought.
But given that most of the code is similar for for the two cases, main difference is in the `validate` function we can most  most probably provided a base class and extend upon it.